Packs/PrismaSaasSecurity/Integrations/SaasSecurityEventCollector/SaasSecurityEventCollector_test.py

```diff
@@ -0,0 +1,297 @@
+                MockedResponse(status_code=200, text=create_events(start_id=901, end_id=1000)),
+                MockedResponse(status_code=200, text=create_events(start_id=1001, end_id=1050)),
+            ],
+            MockedResponse(status_code=200, text=create_events(start_id=1, end_id=1000))
+        ),
+        (
+            300,
+            [
+                MockedResponse(status_code=200, text=create_events(start_id=1, end_id=100)),
+                MockedResponse(status_code=200, text=create_events(start_id=101, end_id=200)),
+                MockedResponse(status_code=200, text=create_events(start_id=201, end_id=280)),
+                MockedResponse(status_code=204)
+            ],
+            MockedResponse(status_code=200, text=create_events(start_id=1, end_id=280))
+        ),
+        (
+            400,
+            [
+                MockedResponse(status_code=200, text=create_events(start_id=1, end_id=100)),
+                MockedResponse(status_code=200, text=create_events(start_id=101, end_id=200)),
+                MockedResponse(status_code=200, text=create_events(start_id=201, end_id=300)),
+                MockedResponse(status_code=204)
+            ],
+            MockedResponse(status_code=200, text=create_events(start_id=1, end_id=300))
+        ),
+    ]
+
+    @pytest.mark.parametrize('max_fetch, queue, expected_events', EVENTS_DATA)
+    def test_fetch_events(self, mocker, mock_client, max_fetch, queue, expected_events):
+        """
+        Given
+           - a queue of responses to fetch events.
+           - max fetch limit
+
+        When -
+            fetching events.
+
+        Then -
+           make sure the correct events are fetched according to the queue and max fetch.
+        """
+        from SaasSecurityEventCollector import fetch_events
+        mocker.patch.object(Client, 'http_request', side_effect=queue)
+        assert expected_events.json().get('events') == fetch_events(client=mock_client, max_fetch=max_fetch)
+
+    @pytest.mark.parametrize('max_fetch, queue, expected_events', EVENTS_DATA)
+    def test_saas_security_get_events(self, mocker, mock_client, max_fetch, queue, expected_events):
+        """
+        Given
+           - a queue of responses to fetch events.
+           - a limit parameter.
+           - a should push events parameter.
+
+        When -
+            executing the get security events command.
+
+        Then
+          - make sure the correct events are fetched according to the queue and max fetch.
+          - make sure in case where there are no events to fetch, a proper message will be returned.
+          - make sure that the send_events_to_xsiam was called in case should_push_events is True
+          - make sure that the send_events_to_xsiam was not called in case should_push_events is False
+        """
+        import SaasSecurityEventCollector
+        import random
+
+        should_push_events = bool(random.choice([True, False]))
```

Comment:
Not a must change, but imo better to put in a 
`for bool in [True,False] `
than to run non-deterministic tests

Reply:
i agree with you, on a second thought, using random is not a good idea, changed it a bit.